osu.Game/Screens/Play/HUD/HitErrorMeters/BarHitErrorMeter.cs

```diff
@@ -233,29 +282,54 @@ public override void OnNewJudgement(JudgementResult judgement)
             if (!judgement.IsHit)
                 return;
 
-            judgementsContainer.Add(new JudgementLine
+            var judgementLine = new JudgementLine(alignment == Anchor.x1)
             {
-                Y = getRelativeJudgementPosition(judgement.TimeOffset),
-                Anchor = alignment == Anchor.x2 ? Anchor.x0 : Anchor.x2,
-                Origin = Anchor.y1 | (alignment == Anchor.x2 ? Anchor.x0 : Anchor.x2),
-            });
-
-            arrow.MoveToY(
-                getRelativeJudgementPosition(floatingAverage = floatingAverage * 0.9 + judgement.TimeOffset * 0.1)
-                , arrow_move_duration, Easing.Out);
+                Anchor = alignment == Anchor.x1 ? Anchor.x1 : (alignment == Anchor.x2 ? Anchor.x0 : Anchor.x2),
+                Origin = Anchor.y1 | (alignment == Anchor.x1 ? Anchor.x1 : (alignment == Anchor.x2 ? Anchor.x0 : Anchor.x2)),
+            };
+
+            if (alignment == Anchor.x1)
+            {
+                judgementLine.X = getRelativeJudgementPosition(judgement.TimeOffset);
+                arrow.MoveToX(
+                    getRelativeJudgementPosition(floatingAverage = floatingAverage * 0.9 + judgement.TimeOffset * 0.1),
+                    arrow_move_duration, Easing.Out);
+            }
+            else
+            {
+                judgementLine.Y = getRelativeJudgementPosition(judgement.TimeOffset);
+                arrow.MoveToY(
+                    getRelativeJudgementPosition(floatingAverage = floatingAverage * 0.9 + judgement.TimeOffset * 0.1)
+                    , arrow_move_duration, Easing.Out);
+            }
+
+            judgementsContainer.Add(judgementLine);
         }
 
-        private float getRelativeJudgementPosition(double value) => (float)((value / maxHitWindow) + 1) / 2;
+        private float getRelativeJudgementPosition(double value) => alignment == Anchor.x1 ? (float)(value / maxHitWindow) / 2 : (float)((value / maxHitWindow) + 1) / 2;
 
         private class JudgementLine : CompositeDrawable
         {
             private const int judgement_fade_duration = 10000;
 
-            public JudgementLine()
+            private readonly bool isVertical;
+
+            public JudgementLine(bool isVertical = false)
             {
-                RelativeSizeAxes = Axes.X;
-                RelativePositionAxes = Axes.Y;
-                Height = 3;
+                this.isVertical = isVertical;
+
+                if (isVertical)
+                {
+                    RelativeSizeAxes = Axes.Y;
+                    RelativePositionAxes = Axes.X;
+                    Width = 3;
+                }
+                else
+                {
+                    RelativeSizeAxes = Axes.X;
+                    RelativePositionAxes = Axes.Y;
+                    Height = 3;
+                }
```

Comment:
These can be fairly simplified with:
```csharp
RelativeSizeAxes = isVertical ? Axes.Y : Axes.X;
RelativePositionAxes = isVertical ? Axes.X : Axes.Y;
Size = isVertical ? new Vector2(3, Size.Y) : new Vector2(Size.X, 3);
```

(making it more easier with what I proposal):
```csharp
RelativeSizeAxes = getAligned(alignment, Axes.X);
RelativePositionAxes = getAligned(alignment, Axes.Y);
Size = getAligned(alignment, new Vector2(Size.X, 3));
```

Reply:
I'm unsure about the utility of the code snippet you commented earlier as it is simply wrapping an inline conditionnal in a function. As for the properties , looks like I forgot to inline them

Reply:
> I'm unsure about the utility of the code snippet you commented earlier as it is simply wrapping an inline conditionnal in a function.

They make the code more readable tbh. I would implement this only if it's possible to and useful for the source code. (forgot to mention that)